colony-common/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use pbc_contract_common::*;
+use rust_decimal::prelude::*;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// A contract type.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum ContractType {
+    Treasury,
+    LightClient,
+    Custom { name: String, description: String },
+}
+
+/// Information of a contract.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct ContractInfo {
+    /// The address of the contract.
+    pub address: String,
+    /// The type of the contract.
+    pub contract_type: ContractType,
+    /// The increasing sequence that is for preventing the replay attack.
+    ///
+    /// A valid message from PBC MUST carry the same number with this,
+    /// in order to succesfully convince the contract. (i.e., this number is something that the consensus should have finalized on-chain).
+    ///
+    /// - Note1: this is totally irrelevant to the account sequence.
+    /// - Note2: the light client contract doesn't need this because the 'block height' provides the same safety guard.
+    pub sequence: u64,
+}
+
+/// An error that can occur when interacting with the contract.
+#[derive(Error, Debug)]
+pub enum Error {
+    /// When there is a problem to access to the full node.
+    #[error("connection error: {0}")]
+    ConnectionError(String),
+    /// When the transaction is rejected by the full node, before it gets to the contract.
+    #[error("transaction rejected: {0}")]
+    TransactionRejected(String),
+    /// When the contract fails to decode the input data.
+    #[error("failed to parse the payload of the transaction")]
+    FailedToParseTransactionPayload,
+    /// When the given proof is invalid.
+    #[error("invalid proof given: got merkle root of {0} but expected {1}")]
+    InvalidProof(String, String),
+    /// When the given message is well decoded and verified, but the message argument is invalid.
+    #[error("invalid message argument given: {0}")]
+    InvalidMessageArgument(String),
+    /// When the relayer account has not enough balance to execute the transaction.
+    #[error("not enough balance: got {0}")]
+    NotEnoughBalance(u64),
+    /// When the account sequence given in the transaction is invalid.
+    #[error("invalid account sequence; expected {0} but got {1}")]
+    InvalidAccountSequence(u64, u64),
+    /// When the contract sequence given in the transaction is invalid.
+    #[error("invalid contract sequence; expected {0} but got {1}")]
+    InvalidContractSequence(u64, u64),
+    /// When the contract fails to execute the transaction with its own error.
+    #[error("internal contract error: {0}")]
+    InternalContractError(String),
+    /// When the contract is missing.
+    #[error("no such contract: {0}")]
+    NoSuchContract(String),
+    /// An unknown error.
```

Comment:
minor 한 거지만, 위에 When~~ 으로 에러 문구 맞출 거라면, 이건 어떨 까요? 반영 안해주셔도 상관 없습니다
```suggestion
    /// When an unknow error occur.
```

Reply:
나중에 에러타입 revisit 하면서 고칠게요